### ISSUE TITLE
@Before tag should have optional scenario parameter

### ISSUE THREAD

**Issue by henningjensen** [cucumber-attic/cuke4duke#11]
When using the @Before to add a setup() you have to write it like this:
    @Before("")
    public void setup(Object scenario) {
    }

In most cases you want this:
    @Before()
    public void setup() {
    }

**Comment by aslakhellesoy** [comment]
No arg to the method can be done. Any suggestions how the @Before tag can take 0 args?

**Comment by henningjensen** [comment]
This is untested but should work:

```
package cuke4duke;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface Before {
    String value() default ""; 
}   
```

**Comment by aslakhellesoy** [comment]
@Before tag should have optional scenario parameter. Closed by f6d0fba9af09f659c8c1893a34e82c354d653f11
